### PR TITLE
upgrade db. index captured_run_id

### PR DESCRIPTION
**Pull request by lakhotiaharshit**
Context: load_by_run_spec was taking much longer time compared to load_by_id or load_by_guid.
Reason: One of the reasons for this is that load_by_run_spec runs sql query over non-indexed columns "captured_run_id"

This PR proposes an upgrade in the database where captured_run_id is also indexed.

 
@jenshnielsen @astafan8

### CODE REVIEW

File: qcodes/dataset/sqlite/db_upgrades/__init__.py [hunk 1]
```diff
@@ -317,3 +317,33 @@ def perform_db_upgrade_7_to_8(conn: ConnectionPlus) -> None:
...
+    pbar = tqdm(range(1), file=sys.stdout)
+    pbar.set_description("Upgrading database; v8 -> v9")
+
+    if n_run_tables == 1:
+        _IX_runs_captured_run_id = """
+                                CREATE INDEX
+                                IF NOT EXISTS IX_runs_captured_run_id
+                                ON runs (captured_run_id DESC)
+                                """
+        with atomic(conn) as conn:
```
Comment (jenshnielsen) [inline qcodes/dataset/sqlite/db_upgrades/__init__.py]: Maybe we should follow codacy here and rename the wrapped conn as something else for clarity

File: qcodes/tests/dataset/test_database_creation_and_upgrading.py [hunk 2]
```diff
@@ -922,3 +923,31 @@ def test_getting_db_version(version):
 
     assert db_v == version
     assert new_v == LATEST_VERSION
+
+
+@pytest.mark.parametrize('db_file',
+                         ['empty',
+                          'some_runs'])
+
```
Comment (astafan8) [inline qcodes/tests/dataset/test_database_creation_and_upgrading.py]: this empty line should not be here, right?

File: qcodes/tests/dataset/test_database_creation_and_upgrading.py [hunk 3]
```diff
@@ -922,3 +923,31 @@ def test_getting_db_version(version):
...
+def test_perform_actual_upgrade_8_to_9(db_file):
+    v8fixpath = os.path.join(fixturepath, 'db_files', 'version9')
```
Comment (astafan8) [inline qcodes/tests/dataset/test_database_creation_and_upgrading.py]: typo?
```suggestion
    v8fixpath = os.path.join(fixturepath, 'db_files', 'version8')
```

File: qcodes/tests/dataset/test_database_creation_and_upgrading.py (hunk 2 above)
Comment (lakhotiaharshit) [reply qcodes/tests/dataset/test_database_creation_and_upgrading.py]: Fixed. Thanks

**Comment by jenshnielsen** [comment]
@astafan8 Any more comments or should we land this?

**Pull request merged**